pyproject.toml

```diff
@@ -9,22 +9,23 @@ build-backend = "setuptools.build_meta"
 [project]
 name = "ndcube"
 description = "A package for multi-dimensional contiguous and non-contiguous coordinate aware arrays."
-requires-python = ">=3.10"
+requires-python = ">=3.12"
```

Comment:
How would people feel about keeping this at 3.11 for one more major release? I'd kinda like to get the ASDF stuff out and not immediately have to bump `dkist` to 3.12.